### ISSUE TITLE
[API] HDB - Add Support for Procedure Multiple OUT Parameters

### ISSUE THREAD

**Issue by ThuF** [SAP/xsk#1360]
Executing a procedure that is expected to return multiple `resultSets`, result into only one `resultSet` being returned.

It seems that the XSJS API is not supporting processing of `OUT` params, hence no multiple `resultSets`:

- https://github.com/SAP/xsk/blob/68a12e49bfde35400fa26944c2d8ad15b57aaae8/modules/api/api-xsjs/src/main/resources/META-INF/dirigible/xsk/hdb/hdb.js#L305-L321

So there are no multiple `resultSets` here:

- https://github.com/SAP/xsk/blob/68a12e49bfde35400fa26944c2d8ad15b57aaae8/modules/api/api-xsjs/src/main/resources/META-INF/dirigible/xsk/hdb/hdb.js#L177-L189

Sample procedure SQL:

```sql
CREATE PROCEDURE "mySchema"."myProcedure"(
    IN firstName NVARCHAR(256),
    IN lastName NVARCHAR(256),
    IN nationalId NVARCHAR(1000),
    OUT outTable "mySchema"."myTableType",
    OUT FLAG TINYINT,
    OUT MESSAGE NVARCHAR(1000),
    OUT TRACEMESSAGE NVARCHAR(1000),
    OUT outTable2 "mySchema"."myTableType2"
) 
...
```

Sample procedure and call can be found at:
- https://www.dirigible.io/api/database/procedure/
- https://www.dirigible.io/samples/basic/database-procedure-call/

**Comment by ThuF** [comment]
**Note:** To be backward compatible with the old XS engine, do uppercase of all `OUT` params:

```sql
CREATE PROCEDURE "mySchema"."myProcedure"(
    IN firstName NVARCHAR(256),
    IN lastName NVARCHAR(256),
    IN nationalId NVARCHAR(1000),
    OUT outTable "mySchema"."myTableType",
    OUT FLAG TINYINT,
    OUT MESSAGE NVARCHAR(1000),
    OUT TRACEMESSAGE NVARCHAR(1000)
) 
...
```

```sql
    OUT outTable "mySchema"."myTableType",
```

->

```sql
    OUT OUTTABLE "mySchema"."myTableType",
```

**Comment by ThuF** [comment]
**Update**: The `OUT` parameters should be registered as follows, before executing the procedure: 

```javascript
connection.prepareCall('"mySchema"."myProcedure"(?,?,?,?,?,?,?)');

...

callableStatement.registerOutParameter(5, java.sql.Types.TINYINT);
callableStatement.registerOutParameter(6, java.sql.Types.NVARCHAR);
callableStatement.registerOutParameter(7, java.sql.Types.NVARCHAR);
```

When building the final result the following calls should be made:

```javascript
callableStatement.getInt(5);
callableStatement.getString(6);
callableStatement.getString(7);
```

**Comment by ThuF** [comment]
**Suggestion:** It might be better to extend the Eclipse Dirigible [Database Procedure API](https://www.dirigible.io/api/database/procedure/) and reuse it, instead of overcomplicating the `$.hdb` XSJS implementation.

**Comment by dpanayotov** [comment]
fixed with #1388
